cmd/aws_signing_helper/main.go

```diff
@@ -252,72 +261,63 @@ func main() {
 			syscall.Exit(1)
 		}
 		var nextRefreshTime time.Time
-		var m sync.Mutex
 		for {
-			if helper.ValidCred(refreshableCred) {
-				nextRefreshTime = refreshableCred.Expiration.Add(-helper.UpdateRefreshTime)
-				fmt.Println("Credentials will be refreshed at ", nextRefreshTime.String())
-				time.Sleep(nextRefreshTime.Sub(time.Now()))
-			}
-
 			credentialProcessOutput, err := helper.GenerateCredentials(&credentialsOptions)
 			if err != nil {
 				log.Fatal(err)
 			}
 
-			//Assign Credential Values
+			// Assign Credential Values
 			refreshableCred.AccessKeyId = credentialProcessOutput.AccessKeyId
 			refreshableCred.SecretAccessKey = credentialProcessOutput.SecretAccessKey
 			refreshableCred.SessionToken = credentialProcessOutput.SessionToken
 			refreshableCred.Expiration, _ = time.Parse(time.RFC3339, credentialProcessOutput.Expiration)
 			if (refreshableCred == helper.TemporaryCredential{}) {
-				log.Fatal("No credentials created")
+				log.Println("no credentials created")
+				syscall.Exit(1)
 			}
 			// Get/Create the AWS Credentials file path
 			awsFile, err := helper.GetOrCreateCredentialsFile()
 			if err != nil {
-				log.Fatal(err)
+				log.Println("unable to get or create AWS credentials file")
+				syscall.Exit(1)
 			}
-			defer awsFile.Close()
 
 			// Read in all profiles in the credentials file
 			var lines []string
-			m.Lock()
 			scanner := bufio.NewScanner(awsFile)
 			for scanner.Scan() {
 				lines = append(lines, scanner.Text())
 			}
-			m.Unlock()
 
 			tmpCredFile, err := ioutil.TempFile("", "aws")
 			if err != nil {
 				log.Fatal(err)
 			}
-			defer tmpCredFile.Close()
-			defer os.Remove(tmpCredFile.Name())
 
-			// write to temporary file
+			// Write to temporary file
 			helper.WriteTo(profile, tmpCredFile, lines, &refreshableCred)
```

Comment:
Tempfile is creating a file in `/tmp` with 0600 permissions meaning that only the process or the whoever the owner the process derived from could access the file. So I don't believe there is any harm in moving credentials out of `~/.aws/credentials` for a very small fragment of time before later removing it. But it may be surprising to some end users who would intuitively think credentials will always remain in `~/.aws`

Reply:
I didn't do anything to change this, but I can add documentation about it in the README if you think it would help.

Reply:
I think it's fine without docs since the switch process is suppose to be transparent and `/tmp` is common in a lot of other software tooling.